### PR TITLE
Add warning message to Gradients and training documentation about `ComplexWarnings`

### PR DESCRIPTION
**Pull request by justinpickering**
**Context:**
- This PR addresses a recurring ComplexWarning issue that appears when using differentiable workflows involving both complex and float types, particularly during backpropagation. Users have reported this issue (see an example [here](https://discuss.pennylane.ai/t/casting-of-complex-to-float-when-using-phaseshift-in-circuit-for-qml/7439)), and [internal discussions](https://xanaduhq.slack.com/archives/C0743CNS9E3/p1730329623028289) have concluded that while the warning is generally harmless, it can still be confusing for users.

**Description of the Change:**
- A warning note has been added to the Gradients and training page in the documentation. The note explains the origin of the ComplexWarning, assures users that it is non-critical in this context, and provides a code snippet to suppress the warning.

**Benefits:**
- Clarifies to users that the ComplexWarning does not indicate a calculation error, which can help avoid unnecessary troubleshooting.
- Offers an easy method to suppress the warning

**Possible Drawbacks:**
- Suppressing warnings could potentially hide other, less common issues involving complex numbers in more advanced workflows
- Users may overlook the warning's informational value if they suppress it without fully understanding the implications

**Related GitHub Issues:**

### CODE REVIEW

File: doc/introduction/interfaces.rst [hunk 1]
```diff
@@ -34,6 +34,17 @@ a :class:`QNode <pennylane.QNode>`, e.g.,
...
+.. warning::
+
+    `ComplexWarning` messages may appear when running differentiable workflows involving both complex and float types, particularly    
+    with certain interfaces. These warnings are common in backpropagation due to the nature of complex casting and do not 
+    indicate an error in computation. If desired, you can suppress these warnings by adding the following code:
+           
+    .. code-block:: python
+
+        import warnings
+        warnings.filterwarnings("ignore", category=ComplexWarning)
```
Comment (mudit2812) [inline doc/introduction/interfaces.rst:46]:
```suggestion
        warnings.filterwarnings("ignore", category=np.ComplexWarning)
```

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -43,6 +43,8 @@
 
 <h3>Documentation 📝</h3>
 
+* Add a warning message to Gradients and training documentation about ComplexWarnings ([#6543](https://github.com/PennyLaneAI/pennylane/pull/6543))
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:46]:
```suggestion
* Add a warning message to Gradients and training documentation about ComplexWarnings
  [(#6543)](https://github.com/PennyLaneAI/pennylane/pull/6543)
```

**Review by mudit2812 — COMMENTED** [review]
Minor suggestions. Don't forget to add your nam to the changelog :)

File: doc/introduction/interfaces.rst (hunk 1 above)
Comment (CatalinaAlbornoz) [reply doc/introduction/interfaces.rst:46]: Should it be `np.` or `pnp.` @mudit2812 ? Or is it the same?

**Review by CatalinaAlbornoz — APPROVED** [review]
Just left a small comment/question about line 46

File: doc/introduction/interfaces.rst (hunk 1 above)
Comment (mudit2812) [reply doc/introduction/interfaces.rst:46]: `np` is probably best. It's general, and suppressing the vanilla numpy warning will work regardless of whether you're using vanilla numpy, jax numpy, or pennylane numpy

**Review by mudit2812 — APPROVED** [review]
